### DISCUSSION TITLE
Qual Templating Engine usar?!

### DISCUSSION THREAD

**Discussion by Cleysonlb** [frontendbr/forum#166] [Discussão]
Fala galera blz?! To querendo implementar no meu workflow um templating engine tipo Jade, handlebars etc... Mas to na duvida de qual usar, a resposta deve ser o que mais se adapta ao projeto ou ao meu workflow, mas decidi trazer para o forum essa discussão :)

**Comment by fdaciuk** [comment]
Só escolher um estilo xD
- O Jade (agora [pug](https://github.com/pugjs/pug)) é baseado em indentação, tipo o Stylus;
- [EJS](http://www.embeddedjs.com/) e [Nunjucks](https://mozilla.github.io/nunjucks/) são mais próximos do HTML. O código fica mais parecido com PHP + HTML xD

Esses são os que eu acho mais completos. Os outros (como handlebars) são mais simples pra fazer estruturas condicionais, if, import, etc).

**Comment by fdaciuk** [comment]
Isso falando de gerar no backend.. se for pra usar um template engine no frontend, vai de React (ou [derivados](https://github.com/developit/preact)). =)

**Comment by felipefialho** [comment]
@Cleysonlb Particulamente no meu workflow o Jade funciona perfeitamente e lindamente, já que estou muito adaptado ao Stylus.

Mas depende bem do que você prefere. Em geral, tanto o Jade como o EJS ou Nunjucks são bem poderosos e oferecem features parecidas.

**Comment by tcelestino** [comment]
Eu uso bastante o Handlebars e como estou "muito" no mundo do Backbone, o lodash ou o underscore.
